### PR TITLE
Implement powermod for negative exponents

### PR DESCRIPTION
**Pull request by anthonyclays**
As I suggested in #15350, allowing `powermod` to take negative exponents is both a natural and useful addition to the function, so I took a stab at implementing this.
Thoughts?

### CODE REVIEW

**Comment by tkelman** [comment]
Looks like there are a few places in `test/numbers.jl` that need to be updated here as well.

**Comment by anthonyclays** [comment]
I updated all other appearances of `powermod`. Perhaps there should be some more tests for BigInts as well?

File: test/numbers.jl [hunk 1]
```diff
@@ -2359,8 +2359,8 @@ end
 @test_throws DivideError powermod(1,0,0)
 @test_throws DivideError powermod(1,0,big(0))
 # negative power domain error
```
Comment (tkelman) [inline test/numbers.jl]: comment's not so accurate any more

**Comment by tkelman** [comment]
> Perhaps there should be some more tests for BigInts as well?

Sure! Improved test coverage is always welcome.

**Comment by anthonyclays** [comment]
The Travis failure is due to a time limit exceeded. However, it still successfully ran the [`intfuncs` tests](https://travis-ci.org/JuliaLang/julia/jobs/121081627#L2115).

Because gmp has [built-in support](https://gmplib.org/manual/Integer-Exponentiation.html#Integer-Exponentiation) for negative exponents in the exact way I described above, I simply removed the negative number check from `powermod(::BigInt, ::BigInt, ::BigInt)` (1cd5ddc)

I also added a number of tests for powermod with BigInt arguments.

**Comment by StefanKarpinski** [comment]
Thanks, @anthonyclays! The OS X CI job timed out so I've restarted it just to be thorough, but it seems unlikely that one will fail when all the rest have passed.

**Comment by anthonyclays** [comment]
Travis timed out again.

**Comment by yuyichao** [comment]
Restarting the travis doesn't make it run on a new (merge) commit so it still runs all the tests on OSX. Please rebase on current master and force push.

**Comment by anthonyclays** [comment]
The x86_64 linux build failed on Travis:
```
From worker 4:	  Test threw an exception of type ReadOnlyMemoryError
From worker 4:	  Expression: rationalize(Int64,prevfloat(0.1)) == 1 // 10
From worker 4:	  ReadOnlyMemoryError()
```
This seems unrelated... ?

**Comment by anthonyclays** [comment]
Third time's the charm for Travis, apparently. Can this be merged?

**Pull request merged**